src/components/WorkPoolAverageLateTime.vue

```diff
@@ -0,0 +1,58 @@
+<template>
+  <span class="work-pool-average-late-time" :class="classes">
+    <template v-if="averageLateness">
+      {{ averageLateness }}
+    </template>
+    <template v-else>
+      N/A
+    </template>
+  </span>
+</template>
+
+<script lang="ts" setup>
+  import { useSubscription } from '@prefecthq/vue-compositions'
+  import { computed } from 'vue'
+  import { useWorkspaceApi } from '@/compositions'
+  import { FlowRunsFilter, WorkPool } from '@/models'
+  import { secondsToApproximateString } from '@/utilities'
+
+  const props = defineProps<{
+    workPool: WorkPool,
+    filter?: FlowRunsFilter,
+  }>()
+
+  const api = useWorkspaceApi()
+
+  const subscriptionOptions = {
+    interval: 30000,
+  }
+
+  const lateFlowRunsFilter = computed<FlowRunsFilter>(() => ({
+    ...props.filter,
+    workPools: {
+      name: [props.workPool.name],
+    },
+    flowRuns: {
+      ...props.filter?.flowRuns,
+      state: {
+        name: ['Late'],
+      },
+    },
+  }))
+
+  const averageLatenessSubscription = useSubscription(api.flowRuns.getFlowRunsAverageLateness, [lateFlowRunsFilter], subscriptionOptions)
+  const averageLateness = computed(() => averageLatenessSubscription.response
+    ? secondsToApproximateString(averageLatenessSubscription.response)
+    : null,
+  )
```

Comment:
nit: You don't really need to format this here and use a ternary. You can just return the response directly and in the template where you're already checking if there is lateness you can format the seconds like `{{ secondsToApproximateString(averageLateness) }}`

Reply:
Ahh true.. might as well cleanup the extra ternary